app/src/main/kotlin/com/softteco/template/utils/BluetoothHelperImpl.kt

```diff
@@ -25,14 +25,19 @@ import com.softteco.template.Constants.INDEX_OF_HUMIDITY
 import com.softteco.template.Constants.START_INDEX_OF_BATTERY
 import com.softteco.template.Constants.START_INDEX_OF_TEMPERATURE
 import com.softteco.template.MainActivity
-import com.softteco.template.data.bluetooth.DataLYWSD03MMC
+import com.softteco.template.data.bluetooth.BluetoothHelper
+import com.softteco.template.data.bluetooth.entity.BluetoothState
+import com.softteco.template.data.bluetooth.entity.DataLYWSD03MMC
 import no.nordicsemi.android.support.v18.scanner.BluetoothLeScannerCompat
 import no.nordicsemi.android.support.v18.scanner.ScanCallback
 import no.nordicsemi.android.support.v18.scanner.ScanResult
 import timber.log.Timber
 import java.util.UUID
+import javax.inject.Inject
+import javax.inject.Singleton
 
-object BluetoothHelper {
+@Singleton
+internal class BluetoothHelperImpl @Inject constructor() : BluetoothHelper {
```

Comment:
`characteristic.value` seems to have been deprecated. Shouldn't we use something another instead?

Reply:
It is recommended not to change based on this: https://stackoverflow.com/questions/76166150/android-ble-characteristics-getvalue-was-deprecated-in-api-level-33-whats-corr